### PR TITLE
User changing its own password

### PR DESCRIPTION
**Pull request by tonijurjevic96**
Allow users to change their own password

Usage: SET PASSWORD TO newPassword REPLACE oldPassword;

### CODE REVIEW

**Comment by tonijurjevic96** [comment]
@kgolubic

**Comment by kgolubic** [comment]
As RN I will use:

- Added functionality that allows each user to change their own password. 
[#2097](https://github.com/memgraph/memgraph/pull/2097)

File: tests/e2e/user_changing_its_own_password/common.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+
+@pytest.fixture
+def connect(**kwargs) -> mgclient.Connection:
+    connection = mgclient.connect(host="localhost", port=7687, **kwargs)
+    connection.autocommit = True
+    cursor = connection.cursor()
+    execute_and_fetch_all(cursor, "USE DATABASE memgraph")
+    try:
+        execute_and_fetch_all(cursor, "DROP DATABASE clean")
```
Comment (Josipmrden) [inline tests/e2e/user_changing_its_own_password/common.py:31]: why do we need this database "clean"?

File: tests/e2e/user_changing_its_own_password/user_changing_its_own_password.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        with driver.session() as session:
+            session.run(query)
+
+
+def test_create_user():
+    AUTH = ("", "")
+    execute_query("CREATE USER IF NOT EXISTS testuser IDENTIFIED BY 'testpassword';", AUTH)
+    AUTH = ("testuser", "testpassword")
+    execute_query("SET PASSWORD TO 'newPassword' REPLACE 'testpassword';", AUTH)
+    print("Created user.")
```
Comment (Josipmrden) [inline tests/e2e/user_changing_its_own_password/user_changing_its_own_password.py:33]: remove print, you can type just assert True

File: tests/e2e/user_changing_its_own_password/user_changing_its_own_password.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        # execute_query("MATCH n RETURN n LIMIT 1;")
+    except AuthError as e:
+        print("Authentication failed")
+
+
+def test_right_credentials():
+    try:
+        AUTH = ("testuser", "newPassword")
+        execute_query("SHOW DATABASE SETTINGS;", AUTH)
+        # execute_query("MATCH n RETURN n LIMIT 1;")
```
Comment (Josipmrden) [inline tests/e2e/user_changing_its_own_password/user_changing_its_own_password.py:52]: remove this execute_query comments

File: tests/e2e/user_changing_its_own_password/user_changing_its_own_password.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    print("Created user.")
+
+
+def test_wrong_credentials():
+    try:
+        AUTH = ("testuser", "testpassword")
+
+        execute_query("SHOW DATABASE SETTINGS;", AUTH)
+
+        # execute_query("SET PASSWORD TO 'newPassword' REPLACE 'testpassword';")
```
Comment (Josipmrden) [inline tests/e2e/user_changing_its_own_password/user_changing_its_own_password.py:42]: remove comments

File: src/glue/auth_handler.cpp [hunk 5]
```diff
@@ -16,6 +16,7 @@
 #include <fmt/format.h>
 
 #include "auth/auth.hpp"
+#include "auth/crypto.hpp"
```
Comment (Josipmrden) [inline src/glue/auth_handler.cpp:19]: see if this header is necessary

**Comment by tonijurjevic96** [comment]
Please delete either the [master < EPIC] or [master < Task] part, depending on what are your needs.

[master < Epic] PR
- [x] Write E2E tests
- [x] Compare the [benchmarking results](https://bench-graph.memgraph.com/) between the master branch and the Epic branch
- [x] Provide the full content or a guide for the final git message
    - Added functionality that allows each user to change its own password


### CI Testing Labels
Please select the appropriate CI test labels _(CI -build=**build-name** -test=**test-suite**)_


### Documentation checklist
- [x] Add the documentation label tag
- [x] Add the bug / feature label tag
- [x] Add the milestone for which this feature is intended
    - If not known, set for a later milestone
- [x] Write a release note, including added/changed clauses
    - Added functionality that allows each user to change its own password
      Usage:
      SET PASSWORD TO newPassword REPLACE oldPassword; 
- [x] Link the documentation PR here
    - https://github.com/memgraph/documentation/pull/819
- [x] Tag someone from docs team in the comments

**Pull request merged**